php/context/class-fieldmanager-context-storable.php

```diff
@@ -52,9 +52,18 @@ protected function save( $data = null ) {
 			$this->save_field( $this->fm, $data, $this->fm->data_id );
 		} else {
 			if ( null === $data ) {
-				$data = isset( $_POST[ $this->fm->name ] ) ? $_POST[ $this->fm->name ] : ''; // WPCS: input var okay. CSRF okay. Sanitization okay.
+				$data = isset( $_POST[ $this->fm->name ] ) ? $_POST[ $this->fm->name ] : array(); // WPCS: input var okay. CSRF okay. Sanitization okay.
+
+				// Multiple select and radio fields with no values chosen are left out of the post request altogether.
+				foreach ( $this->fm->children as $child ) {
+					$is_multi_select = ( 'select' == $child->field_class ) && ( true === $child->multiple );
+					$is_radio = ( 'radio' == $child->field_class );
+					if ( ( $is_multi_select || $is_radio ) && ! isset( $data[ $child->name ] ) && ! $child->skip_save ) {
```

Comment:
It looks like `skip_save` is also checked in `\Fieldmanager_Context_Storable::save_field()`. Is it needed here?